src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+import "./BloodPressureVisualization.css";
+import { useEffect, useState } from "react";
+import MyDataHelps from "@careevolution/mydatahelps-js";
+import React from "react";
+import { Button, DateRangeNavigator, LoadingIndicator, Title } from "../../presentational";
+import DumbbellChart from "../../presentational/DumbbellChart";
+import { Dumbbell, ClosedInterval, Axis, DataPoint, DumbbellClass } from "../../presentational/DumbbellChart/DumbbellChart";
+import { addDays, format, isEqual, isMonday, previousMonday, startOfDay } from "date-fns";
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import { BloodPressureDataParameters, BloodPressureDataPoint, BloodPressureDataProvider } from "../../../helpers/blood-pressure-data-providers/query-blood-pressure";
+
+export enum Category {"Low", "Normal", "Elevated", "Stage 1", "Stage 2", "Crisis", "Unknown"};
+export type BloodPressurePreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface BloodPressureVisualizationProps {
+    previewState? : BloodPressurePreviewState,
+    previewDataProvider : BloodPressureDataProvider,
+    dataParameters : BloodPressureDataParameters
+};
+
+
+interface BloodPressureDumbbell {
+    date: Date,
+    dumbbell : Dumbbell,
+    averageSystolic : number,
+    averageDiastolic : number
+};
+
+interface BloodPressureMetrics {
+    averageSystolic? : number,
+    averageSystolicAlert? : string,
+    averageSystolicAlertClass : string,
+    averageDiastolic? : number,
+    averageDiastolicAlert? : string,
+    averageDiastolicAlertClass : string,
+    maxSystolic? : number,
+    maxSystolicAlert? : string,
+    maxSystolicAlertClass : string,
+    minDiastolic? : number, 
+    minDiastolicAlert? : string,
+    minDiastolicAlertClass : string
+}
+
+export default function (props : BloodPressureVisualizationProps) {
+    const _minDiastolic = 0;
+    const _maxSystolic = 250; 
+    const yInterval : ClosedInterval = {values: [_minDiastolic, _maxSystolic]};
+    const axis : Axis = {yRange : yInterval, yIncrement : 50, xIncrement : (85/7)};
+
+    const [loadingData, setLoadingData] = useState(false);
+    const [bpData, setBpData] = useState<BloodPressureDumbbell[]>([]);
+    const [weeklyDataForViz, setWeeklyDataForViz] = useState<Dumbbell[]>([]);
+    const [startOfWeek, setStartOfWeek] = useState<Date>(initializeWeekPager());
+    const [metrics, setMetrics] = useState<BloodPressureMetrics>();
+
+    async function initialize(){
+
+        if (!loadingData){
+            setLoadingData(true);
+
+            if (props.previewState && props.previewState == "Loading"){
+                return;
+            }
+            else 
+            {
+                props.previewDataProvider(props.dataParameters).then((bloodPressureDataPoints) => {
```

Comment:
I think I am confused by the data provider setup here...why is this called the preview data provider, is it not supplying real data in a real use case?

In any case, I think as mentioned before I think the props can just have a SurveyBloodPressureDataParameters and we can just call the SurveyBloodPressureDataProvider from here